src/de/dkfz/b080/co/knowledge/metadata/COMetadataAccessor.groovy

```diff
@@ -64,15 +64,27 @@ class COMetadataAccessor {
 
     }
 
-    protected String extractSampleNameFromBamBasename(String filename, boolean enforceAtomicSampleName) {
-        String[] split = filename.split(StringConstants.SPLIT_UNDERSCORE)
-        if (split.size() <= 2) {
-            return null
-        }
-        String sampleName = split[0]
-        if (!enforceAtomicSampleName && split[1].isInteger() && split[1].length() <= 2)
-            sampleName = split[0..1].join(StringConstants.UNDERSCORE)
-        return sampleName
+    // TODO: Regex?
+    String extractSampleNameFromBamBasename(String filename, ExecutionContext context) {
+        COConfig cfg = new COConfig(context)
+
+        // Get list of all known samples. Sort and revert them, so we can use them properly.
+        // Get rid of empty samples! Sort and revert for first match searches. (e.g. control_abc comes before control!)
+        def listOfAll = (cfg.possibleControlSampleNamePrefixes + cfg.possibleTumorSampleNamePrefixes)
+                .findAll().sort().reverse()
```

Comment:
is there already a toLowerCase() somewhere in preprocessing of SampleNamePrefixes or should we add it here? (cf. line 81)

Reply:
Please take a look at the test cases and extend them with your ideas!